satpy/scene.py

```diff
@@ -1081,12 +1085,106 @@ def to_xarray_dataset(self, datasets=None):
         ds.attrs = mdata
         return ds
 
+    def to_xarray(self,
+                  datasets=None,  # DataID
+                  header_attrs=None,
+                  exclude_attrs=None,
+                  flatten_attrs=False,
+                  pretty=True,
+                  include_lonlats=True,
+                  epoch=None,
+                  include_orig_name=True,
+                  numeric_name_prefix='CHANNEL_'):
+        """Merge all xr.DataArray(s) of a satpy.Scene to a CF-compliant xarray object.
```

Comment:
**What if I want an xarray object version of the Scene, but no CF changes?** 
You just do `scn[your_variable_of_interest]`

What if I want to convert the Scene's contents to CF-compliant DataArrays?
The implemented `make_cf_datarray` does the job ;)

Reply:
`scn[your_variable]` doesn't convert the entire Scene to a Dataset or to a DataTree.

Reply:
Ok. If you agree @djhoese, in a following PR, we can add an argument like `cf=True` in `collect_cf_datasets` that will do the job.
In the same spirit, we might want to ask if the 'area' attributes should be dropped, and/or the attribute serialized!

Reply:
Interesting idea. I guess I'm ok with a future PR, but I'm a little disappointment that htis deprecation warning will remain in `to_xarray_dataset` when the current functionality is not the same. I think I'd prefer removing the deprecation warning for now until this can be done.

And adding a boolean `cf=True` to a function with `_cf_` in the name doesn't sound like a good idea. These seem like separate but maybe intertwined behaviors. Converting the Scene to a Dataset or DataTree (my PR) and converting a collection of DataArrays to something CF-friendly. Yes, with optional serialization...somehow.

Reply:
I agree with all that you say. I remove the deprecation warning to `xarray_dataset` and in one of the next PR, I noted down to add the arguments `cf=True, serialize=True, drop_area=True` to `to_xarray`!